### PR TITLE
Enable jenkins testing on AIX71

### PR DESCRIPTION
**Pull request by AshCripps**
Start to enable testing to be done on the new AIX71 machines.

I cant remember if @sam-github added the label to node-test-commit-aix or not

also does `anyType` cover _both_ test and release or just test?

### CODE REVIEW

**Comment by richardlau** [comment]
> I cant remember if @sam-github added the label to node-test-commit-aix or not

I don't think that has been done as otherwise builds would be running on the AIX 7.1 machines now.

> also does `anyType` cover _both_ test and release or just test?

It covers both. Or more technically it covers any setting of `buildType`: https://github.com/nodejs/build/blob/6bcc28ea91d2c767533869bcc264bb544a839b52/jenkins/scripts/VersionSelectorScript.groovy#L4-L5

**Comment by AshCripps** [comment]
> I don't think that has been done as otherwise builds would be running on the AIX 7.1 machines now.

Yeah I checked our internal checklist and it was the next step for him to do.


> It covers both. Or more technically it covers any setting of `buildType`:

Thought so.

**Comment by richardlau** [comment]
> > > also does `anyType` cover both test and release or just test?
> >
> > It covers both. Or more technically it covers any setting of `buildType`:
> 
> Thought so.

Furthermore currently the only type checked is `releaseType` (`buildType` is `release`). @joaocgreis is planning some changes (https://github.com/nodejs/build/issues/2084#issuecomment-564098021) that will also add a `testType` if we want to have different rules for test only.

File: jenkins/scripts/VersionSelectorScript.groovy [hunk 1]
```diff
@@ -61,6 +61,7 @@ def buildExclusions = [
 
   // AIX61 -------------------------------------------------
   [ /aix61/,                          anyType,     lt(6)   ],
+  [ /aix71/,                          anyType,     lt(14)   ],
```
Comment (sam-github) [inline jenkins/scripts/VersionSelectorScript.groovy]: nit: there appears to be one extra space in this line...

**Comment by sam-github** [comment]
@AshCripps I think you might not have noticed:
> I think we also need to add a line to exclude RELEASE from aix71 for lt(15) (everything)

**Comment by richardlau** [comment]
> 
> 
> @AshCripps I think you might not have noticed:
> 
> > I think we also need to add a line to exclude RELEASE from aix71 for lt(15) (everything)

We don't want to build 14 on AIX 7.1?

**Comment by sam-github** [comment]
@richardlau yes, we _want_ to, but we do this in order, usually:
```
for each V in (14, 13, 12, 10)
  enable test builds for V on NEW MACHINE
  wait for tests to be green
  stop release builds on OLD MACHINE
  start release builds on NEW MACHINE
  // setup in .groovy must ensure that ONLY one machine gets selected to do release builds on a V
  wait for no reports of problems for users
end
```

I don't know, I guess we cloud start doing testing and releases on the 7.1 machines for _all_ release lines `> = 10` immediately, but we're generally more cautious.

I pushed a fixup commit.

@richardlau LGTY with above context? I could add the aix61 build label to node-test-commit-aix as soon as this lands.

**Comment by richardlau** [comment]
SGTM. Thanks for the context.

**Comment by AshCripps** [comment]
> @AshCripps I think you might not have noticed:
> 
> > I think we also need to add a line to exclude RELEASE from aix71 for lt(15) (everything)

Sorry @rsam completely blanked that thanks for the catch!

**Pull request merged**